src/rio.js

```diff
@@ -238,78 +241,89 @@ class View {
+        },
+        onNodeDiscarded: node => {
+          // unmount our associated instance
+          if (isElement(node) && node.hasAttribute('data-rio-id')) {
+            const rioId = node.getAttribute('data-rio-id');
+            const instance = registry.idInstances[rioId];
+            instance.dispatch('unmount');
+            instance.unmount();
           }
-          instance._setEl(node);
-        }
-      },
-      onNodeAdded: node => {
-        // mount our view instance if it is new
-        if (isElement(node) && node.hasAttribute('data-rio-view') && node.hasAttribute('data-rio-id')) {
-          const rioId = node.getAttribute('data-rio-id');
-          const instance = registry.idInstances[rioId];
-          instance.el = node;
-          if (!instance._mounted) {
-            instance.dispatch('mount');
+        },
+        onBeforeNodeDiscarded: node => {
+          // don't remove elements with rio-sacrosanct attribute
+          return isElement(node) && !node.hasAttribute('rio-sacrosanct');
+        },
+        onBeforeElUpdated: ( fromNode, toNode ) => {
+          if (toNode.hasAttribute('data-rio-should-render-false')) {
+            return false;
+          }
+          if (isElement(fromNode) && document.activeElement == fromNode && fromNode.hasAttribute('rio-uninterruptable-input')) {
+            // don't update the focused element if it is uninterruptable
+            return false;
+          }
+        },
+        onBeforeNodeAdded: node => {
+          // transplant existing view instance to its new element if need be
+          if (isElement(node) && node.hasAttribute('data-rio-id')) {
+            const rioId = node.getAttribute('data-rio-id');
+            const instance = registry.idInstances[rioId];
+            // if the old element is in the dom, remove it and throw it away
+            if (instance && instance.el && instance.el.parentNode) {
+              instance.el.parentNode.removeChild(instance.el);
+            }
+            instance._setEl(node);
+          }
+        },
+        onNodeAdded: node => {
+          // mount our view instance if it is new
+          if (isElement(node) && node.hasAttribute('data-rio-view') && node.hasAttribute('data-rio-id')) {
+            const rioId = node.getAttribute('data-rio-id');
+            const instance = registry.idInstances[rioId];
+            instance.el = node;
+            if (!instance._mounted) {
+              instance.dispatch('mount');
+              instance._mounted = true;
+            }
           }
         }
+      });
+
+      while (this._updatedQueue.length) {
+        const view = this._updatedQueue.shift();
+        view.dispatch('updated');
       }
-    });
 
-    while (this._updatedQueue.length) {
-      const view = this._updatedQueue.shift();
-      view.dispatch('updated');
+      this.dispatch('updated');
+    } catch (e) {
+      throw new Error(e);
```

Comment:
Looks like I was wrong here about the stack trace business actually?  If we just throw `e` rather than `new Error(e)`, does that do more like we want?  To take or to leave...

Reply:
Meaning, as it is now, our stack trace shows us back to this line.  But what we'd rather have is the line where the actual error happened...